### DISCUSSION TITLE
Why was `addressOrName` field removed?

### DISCUSSION THREAD

**Discussion by o-az** [wevm/wagmi#2368] [Q&A] [answered]
I'm curious behind the decision. It was quite handy especially with hooks like `useEnsAvatar`. Pretty much perfect when I'm calling the hook based on the result of user input in a search bar. 

https://wagmi.sh/react/migration-guide#useensavatar-1

**Comment by tmm** [comment, marked as answer]
We sacrificed a bit of convenience for simplicity and performance.

- Allows ENS names to be cached more easily by resolving directly with it's own function
- Don't have to do conditional check if ENS name or address internally in functions
- ENS name-related errors are returned where expected (e.g. `useEnsAddress`) versus throwing out of context (e.g. `useContractWrite`, `useSendTransaction`, etc.)
- There are other types of names besides ENS (CCIP Read). How do you decide what falls into `addressOrName` or not.
